### DISCUSSION TITLE
What's the current state of the plugin?

### DISCUSSION THREAD

**Discussion by Hexadotz** [blackears/cyclopsLevelBuilder#74] [General]
Just checking on its development, its been quite a while and I don't mean to rush you or anything, just wanted to know the situation now.

On another note the lines around a selected object in the editor is very buggy, especially when you rotate the camera they go all over the place then goes back to their designated location

**Comment by blackears** [comment]
I started taking contracts on Fiverr recently which has sucked up nearly all my free time for the last two months.  I have made some progress - the HEAD now has the new CyclopsBlock object that can be anywhere in the scene graph.  The outline bug you're talking about has also been fixed.  However, you'll need to upgrade your project if you start using 1.0.2_dev so upgrading is non-reversable to 1.0.1.

**Reply by Hexadotz** [reply]
This is great! Now it should be more performance friendly
If i may suggest something that will probably improve performance even more, you can make a tool that bake multiple blocks into one this way instead of tracking the collision of each of said blocks the engine only track just one, theoretically this should improve performance but idk...
Also i wish i could've sended some bucks your way you really deserve it but i literally cannot do that since my country doesn't have any means to donate

**Reply by blackears** [reply]
I think that would actually cause things to slow down.  Physics engines can't really handle concave meshes, so if you give it a mesh that is not convex, it will automatically split it into convex pieces.  The blocks are currently all convex, so there's no need for them to be further split.  It's also important for the designer to have some control over the shapes of collision shapes.  For example a very long thin collision box (like a really long wall) would probably not be something you want as a single piece but rather several different pieces.  While having many convex shapes may waste some memory, it should be pretty optimal when it comes to physics processing speed.

**Reply by Hexadotz** [reply]
I see. It makes sense actually and also pinpoint some problems i actually had (the long wall example) so yeah something as long as that do cause wierd physics so i should've done my homework, my bad


I do however have some sort of a replacement for this idea without any loses (hopefully). as im using the plugin i find it hard to keep track of a lot of blocks for an example if i create a house with windows and a doorway that's going to be a lot of boxes, so what if the location of said house isn't good and i need to move it a bit. I'd have to manually select every single mesh then move it which is annoying.
I've been thinking about a grouping features one where you select multiple blocks and then put them all in one group it will definitely be very helpful and speed up the level building process
On a side note the material scaling dosent work

**Comment by blackears** [comment]
I noticed the material scaling stopped working and hope to get to it soon.  Grouping things somehow is also a good idea and something I need to think about.
